src/Sarif/FileRegionsCache.cs

```diff
@@ -274,23 +275,48 @@ private void PopulateCharLength(NewLineIndex lineIndex, Region region)
             {
                 region.CharLength = charLength;
             }
-            SarifUtilities.DebugAssert(region.CharLength == charLength);
+            Assert(region.CharLength == charLength);
         }
 
         private NewLineIndex GetNewLineIndex(Uri uri, out string fileText)
         {
-            fileText = null;
+            Tuple<string, NewLineIndex> entry = _cache[uri.LocalPath];
```

Comment:
>_cache [](start = 48, length = 6)

This is so cool, it makes the entry if it wasn't there before! ... is this how caches like this usually work? It's so automagical I wonder if it's confusing... I only understood it because I'd _just_ read Cache.cs. Never mind, I'm probably overthinking this.

Reply:
Yes, this is the normal way, not my own idea. If you write a cache without this, the caller has to call TryGet and then load the value and then remember to stuff it in with a setter - the "self-loading" design avoids the risk of exactly the bug this class had before (building a cache and never putting anything in). =)